src/snarky.d.ts

```diff
@@ -51,489 +53,767 @@ declare interface ProvablePure<T> extends Provable<T> {
 
   /**
-   * Little endian binary representation of the field element.
+   * Returns an array of {@link Bool} elements representing [little endian binary representation](https://betterexplained.com/articles/understanding-big-and-little-endian-byte-order/) of this {@link Field} element.
+   * 
+   * **Warning: Binary operations on snarkyJS are currently really costy on chain. Do not use this function if it is not absolutely necessary.
+   * 
+   * @return An array of {@link Bool} element representing little endian binary representation of this {@link Field}.
    */
   toBits(): Bool[];
 
   /**
-   * Little endian binary representation of the field element.
-   * Fails if the field element cannot fit in `length` bits.
+   * Returns an array of {@link Bool} elements representing [little endian binary representation](https://betterexplained.com/articles/understanding-big-and-little-endian-byte-order/) of this {@link Field} element.
+   * Throws an error if the element cannot fit in `length` bits.
+   * 
+   * **Warning: Binary operations on snarkyJS are currently really costy on chain. Do not use this function if it is not absolutely necessary.
+   * 
+   * @param length - the number of bits to fit the element. If the element does not fit in `length` bits, the functions throws an error.
+   * 
+   * @return An array of {@link Bool} element representing little endian binary representation of this {@link Field}.
    */
   toBits(length: number): Bool[];
 
   /**
-   * Check if this {@link Field} equals another {@link Field}-like value.
-   * Returns a {@link Bool}.
+   * Check if this {@link Field} is equal another `field-like` value.
+   * Returns a {@link Bool}, which is a `provable` type and can be used to prove the validity of this statement.
    *
    * ```ts
-   * Field(2).equals(2); // Bool(true)
+   * Field(5).equals(5).assertEquals(Bool(true));
    * ```
+   * 
+   * @param value - the `field-like` value to compare with this {@link Field}.
+   * 
+   * @return A {@link Bool} representing if this {@link Field} is equal another `field-like` value.
    */
-  equals(y: Field | number | string | boolean): Bool;
+  equals(value: Field | number | string | boolean): Bool;
 
-  // TODO: Izzy to document
+  /**
+   * **Warning: This function is mainly for internal use. Normally it is not intended to be used by a zkApp developer.
+   * 
+   * In snarkyJS, addition and scaling (multiplication of variables by a constant) of variables is represented as an AST - [abstract syntax tree](https://en.wikipedia.org/wiki/Abstract_syntax_tree). For example, the expression `x.add(y).mul(2)` is represented as `Scale(2, Add(x, y))`.
+   * 
+   * Only when the variable is needed in a multiplicative or any higher level constraint (for example multiplication of two {@link Field} elements) a new internal variable is created to represent the operation.
+   * 
+   * The `seal()` function tells snarkyJS to stop building an AST and create a new variable right away.
```

Comment:
Awesome comment! Just a small thing: Can you spell it SnarkyJS instead of snarkyJS everywhere?
```suggestion
   * The `seal()` function tells SnarkyJS to stop building an AST and create a new variable right away.
```